oslo_vmware/image_transfer.py

```diff
@@ -346,3 +360,58 @@ def upload_image(context, timeout_secs, image_service, image_id, owner_id,
         updater.stop()
         read_handle.close()
     LOG.debug("Uploaded image: %s.", image_id)
+
+
+def image_pull_from_swift(swift_url, **kwargs):
+    (lease, lease_info) = \
+        rw_handles.VmdkWriteHandle._create_import_vapp_lease(
+            kwargs.get('session'), kwargs.get('resource_pool'),
+            kwargs.get('vm_import_spec'), kwargs.get('vm_folder'))
+
+    return image_pull_from_url(
+        kwargs.get('session'), lease, lease_info, swift_url.url,
+        size=kwargs.get('image_size'),
+        ssl_thumbprint=swift_url.ssl_thumbprint(),
+        headers={'X-Auth-Token': swift_url.auth_token})
```

Comment:
I think all the headers should be returned from `swift_url`, so that it is completely encapsulated